update notebook rendering: Bayes Factor

reshamas opened the pull request:
### References
Towards https://github.com/pymc-devs/pymc-examples/issues/56
Towards [#5460](https://github.com/pymc-devs/pymc/issues/5460)
References #394 

### Description
Added header and footer to the file similar to the Jupyter Style Guide.

#DataUmbrellaPyMCSprint

reshamas commented:
@aloctavodia This notebook has you as the author, but I wasn't able to find you in the commit history.  So, I wasn't sure which PR to add as a reference.

OriolAbril commented:
You need to add the section title to the cell with the authors, a level 2 markdown header.

---

Most notebooks were created and added to the pymc repo, if you want to find the PR adding it you'll need to search there. They were then moved here to reduce the size of the pymc repo and package. The updates to v4 (which this one has already had) have all been here and are easier to track, you can check the history of each notebook independently on github. 

It is not necessary for you to track the history of the notebook, especially while on the pymc repo, what is a requirement of new PRs is to preserve any existing authorship mentions and make sure they follow the right formatting and adding your changes there too.

It is also recommended but not required for attribution bullet points to have a link to the PR if you don't know it, it is more important to keep authorship and attribution notes than to add the link

---

You will also need to run pre-commit to get CI to pass. All notebooks have two representations as raw text (myst) and as ipynb file which pre-commit makes sure they are synced

aloctavodia commented:
Hi @reshamas I am not able to find the original PR, it was a looong time ago. Also notice that there have been several updates to this notebook that are listed as the practice of consistenly adding a changelog to the notebook is relatively new. The last change to this notebook was this #342

reshamas commented:
@aloctavodia I was able to find the PRs:
* Authored by Osvaldo Martin in September, 2017 ([pymc#2563](https://github.com/pymc-devs/pymc/pull/2563))
* Updated by Osvaldo Martin in August, 2018 ([pymc#3124](https://github.com/pymc-devs/pymc/pull/3124))
* Updated by Osvaldo Martin in May, 2022 ([pymc-examples#342](https://github.com/pymc-devs/pymc-examples/pull/342))

The latest bottleneck to me moving this PR forward is this error:
```bash
pymc-projects/pymc-devs-repos/pymc-examples  nb_bayes_factor ✗                                         2d ⚑  
▶ sphinx-build examples/ _build -b html
Running Sphinx v4.2.0

Extension error:
Could not import extension myst_nb (exception: No module named 'myst_nb')
(talks) 
```

reshamas commented:
@aloctavodia Where would I find the pre-commit instructions?
>You will also need to run pre-commit to get CI to pass. All notebooks have two representations as raw text (myst) and as ipynb file which pre-commit makes sure they are synced

OriolAbril commented:
If you followed the steps in https://pymc-data-umbrella.xyz/en/latest/sprint/tutorials/environment_setup.html#environment-setup and https://pymc-data-umbrella.xyz/en/latest/sprint/tutorials/pr_tutorial.html pre commit will run automatically when you commit. This can also be a bit slow at times, so it is also perfectly ok to run pre commit manually instead of this automatic approach. There is some guidance on that in the jupyter style guide https://www.pymc.io/projects/docs/en/latest/contributing/jupyter_style.html#pre-commit-and-code-formatting

aloctavodia reviewed (approved):
LGTM!

aloctavodia commented:
Thanks @reshamas
